src/Milestones/ui/Root.tsx

```diff
@@ -0,0 +1,33 @@
+import { IPlayer } from "../../PersonObjects/IPlayer";
+import { Milestones } from "../Milestones";
+import { Milestone } from "../Milestone";
+import * as React from "react";
+
+type IProps = {
+    player: IPlayer;
+}
+
+function highestMilestone(p: IPlayer, milestones: Milestone[]): number {
+    let n = -1;
+    for(let i = 0; i < milestones.length; i++) {
+        if(milestones[i].fulfilled(p)) n = i;
+    }
+
+    return n;
+}
+
+export function Root(props: IProps) {
+    const n = highestMilestone(props.player, Milestones);
+    return (<>
+        <h1>Milestones</h1>
+        <p>Milestones don't reward you for completing them. They are here to guide you if you're lost. They will reset when you install Augmentations.</p><br />
+        
+        <h2>Completing fl1ght.exe</h2>
+        <li>
+            {Milestones.map((milestone: Milestone, i: number) => {
```

Comment:
[This can be extracted out to make it more readable. ](https://reactjs-bot.github.io/react/docs/lists-conditional-rendering.html#rendering-multiple-components)